File: frontend/src/types/part_lot.ts
export interface PartLot {
  id: number;
  component_id: number;
  location_id: number | null;
  quantity: number;
  description: string | null;
  expiry_date: string | null;
  needs_refill: boolean;
  location_name: string | null;
  created_at: string;
}

export interface PartLotCreate {
  location_id?: number;
  quantity: number;
  description?: string;
  expiry_date?: string;
  needs_refill?: boolean;
}

export interface PartLotUpdate extends Partial<PartLotCreate> {}

File: backend/app/schemas/part_lot.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class PartLotBase(BaseModel):
    location_id: Optional[int] = None
    quantity: int = 0
    description: Optional[str] = None
    expiry_date: Optional[date] = None
    needs_refill: bool = False


class PartLotCreate(PartLotBase):
    pass


class PartLotUpdate(BaseModel):
    location_id: Optional[int] = None
    quantity: Optional[int] = None
    description: Optional[str] = None
    expiry_date: Optional[date] = None
    needs_refill: Optional[bool] = None


class PartLotResponse(PartLotBase):
    id: int
    component_id: int
    location_name: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)
